apps/web-app/src/app/features/links/state/delete-link/delete-link.epic.ts

```diff
@@ -0,0 +1,33 @@
+import {ArchivedLink, PendingLink} from '@foxy-news/links';
+import {Epic} from 'redux-observable';
+import {from, of} from 'rxjs';
+import {mergeMap, switchMap, withLatestFrom} from 'rxjs/operators';
+import {AppState} from '../../../../state/app-state';
+import {firebaseApp} from '../../../firebase/firebase.app';
+import {deleteLinkAction} from './delete-link.action';
+
+const firestore = firebaseApp.firestore();
+
+export const deleteLinkEpic: Epic<ReturnType<typeof deleteLinkAction>, any, AppState> = (action$, state$) =>
+  action$.ofType(deleteLinkAction.type).pipe(
+    withLatestFrom(state$),
+    mergeMap(([action, state]) => {
```

Comment:
co nam tutaj daje mergeMap?

Reply:
W przypadku switchMap gdyby przyszła kolejna akcja przez zakończeniem obsługi pierwszej, odsubskrybowalibyśmy się od observable z pierwszego callbacku. Gdyby w callbacku było angularowe zawołanie http, to byśmy je anulowali. Nie wiem jak zachowa się w takim przypadku firebase.
 
Nie zależy nam na tym, żeby zawsze wykonywana była tylko jedna operacja na raz. Jak przyjdą dwie akcje usunięcia linku, to mają się usunąć obie. Bezpieczniej jest użyć mergeMap

Reply:
no i super :+1: